bip-undef-0.mediawiki

```diff
@@ -0,0 +1,113 @@
+<pre>
+  BIP: ???
+  Title: Peer Authentication
+  Author: Jonas Schnelli <dev@jonasschnelli.ch>
+  Status: Draft
+  Type: Standards Track
+  Created: 2016-03-23
+</pre>
+
+== Abstract ==
+
+This BIP describes a way that peers can authenticate to other peers to guarantee ownership and/or allowing to access additional or limited services.
+
+== Motivation ==
+
+We assume peer operators want to limit the access of different services to a selective subset of peers. Also we assume peers want to connect to specific peers to broadcast or filter transactions (or similar action that reveals sensitive information’s) and therefore they want to authenticate the remote peer and make sure that they have not connected to a MITM. 
+
+Benefits with peer authentication:
+* Nodes could allow resource hungry transaction filtering only to specific nodes
+* Nodes could allow access to sensitive information that can lead to node fingerprinting (fee estimation)
+* Nodes could be sure they do not connect to a MITM
+* Nodes could allow custom message types (private extensions) to authenticated peers
+
+A simple authentication scheme based on elliptic cryptography will allow peers to identify each other and selective allow access to restricted services or reject the connection if the identity could not be verified.
+
+== Specification ==
+
+The authentication scheme proposed in this BIP uses ECDSA, '''secrets will never be transmitted'''.
+
+'''Authentication initialization must only happen if encrypted channels has been established (according to BIP-???? [1]).'''
+
+The '''encryption-session-ID''' is available once channels are encrypted (according to BIP-???? [1]).
+
+Authentication can be verified by verifying the signature of the encryption-session-ID with a provided "identity-pubkey" (standard ec pubkey).
+
+The authentication state must be kept until the encryption/connection terminates.
+
+=== Known-peers and authorized-peers database ===
+Each peer that supports p2p authentication must provide two user editable databases
+
+# '''known-peers''' contain known identity-pubkeys together with a network identifier (IP & port), similar to the "known-host" file supported by openssh.
+# '''authorized-peers''' contains identity-pubkeys, together with allowed services (example: encryption,filtering,fee-estimation).
```

Comment:
Nit: Can we add a space after each comma?
encryption, filtering, fee-estimation